app/components/entity_mentions/comments/fetching.py

```diff
@@ -0,0 +1,209 @@
+}
+EVENT_COLOR = 0x3498DB  # blue
+ENTITY_UPDATE_EVENTS = frozenset({"closed", "locked", "merged", "reopened", "unlocked"})
+SUPPORTED_EVENTS = {
+    "assigned": "Assigned `{event.assignee.login}`",
+    "labeled": "Added the `{event.label.name}` label",
+    "milestoned": "Added this to the `{event.milestone.title}` milestone",
+    "review_requested": "Requested review from `{reviewer}`",
+    "unassigned": "Unassigned `{event.assignee.login}`",
+    "unlabeled": "Removed the `{event.label.name}` label",
+}
+
+
+class CommentCache(TTRCache[tuple[EntityGist, str, int], Comment]):
+    async def fetch(self, key: tuple[EntityGist, str, int]) -> None:
+        entity_gist, event_type, event_no = key
+        coro = {
+            "discussioncomment-": get_discussion_comment,
+            "issuecomment-": _get_issue_comment,
+            "pullrequestreview-": _get_pr_review,
+            "discussion_r": _get_pr_review_comment,
+            "event-": _get_event,
+            "discussion-": _get_entity_starter,
+            "issue-": _get_entity_starter,
+        }.get(event_type)
+        if coro is None:
+            return
+        with suppress(RequestFailed):
+            self[key] = await coro(entity_gist, event_no)
+
+
+comment_cache = CommentCache(1800)  # 30 minutes
+
+
+def _make_author(user: BaseModel | None) -> GitHubUser:
+    if not user:
+        return FALLBACK_AUTHOR
+    return GitHubUser(**user.model_dump())
+
+
+async def _get_issue_comment(entity_gist: EntityGist, comment_id: int) -> Comment:
+    owner, repo, _ = entity_gist
+    comment_resp, entity = await asyncio.gather(
+        gh.rest.issues.async_get_comment(owner, repo, comment_id),
+        entity_cache.get(entity_gist),
+    )
+    comment = comment_resp.parsed_data
+    return Comment(
+        author=_make_author(comment.user),
+        body=cast(str, comment.body),
+        entity=entity,
+        entity_gist=entity_gist,
+        created_at=comment.created_at,
+        html_url=comment.html_url,
+    )
+
+
+async def _get_pr_review(entity_gist: EntityGist, comment_id: int) -> Comment:
+    comment = (
+        await gh.rest.pulls.async_get_review(*entity_gist, comment_id)
+    ).parsed_data
+    return Comment(
+        author=_make_author(comment.user),
+        body=comment.body,
+        entity=await entity_cache.get(entity_gist),
+        entity_gist=entity_gist,
+        created_at=cast(dt.datetime, comment.submitted_at),
+        html_url=comment.html_url,
+        color=STATE_TO_COLOR.get(comment.state),
+        kind="Review",
+    )
+
+
+def _prettify_suggestions(comment: PullRequestReviewComment) -> str:
+    suggestions = [
+        c for c in extract_codeblocks(comment.body) if c.lang == "suggestion"
+    ]
+    body = comment.body
+    if not suggestions:
+        return body
+
+    start = cast(int | None, comment.original_start_line)
+    end = cast(int, comment.original_line)
+    hunk_size = end - (end if start is None else start) + 1
```

Comment:
`start` can't be 0 right?  (Line 0 logically doesn't make sense to me, but who knows.)
Subjective.
```suggestion
    hunk_size = end - (start or end) + 1
```

Reply:
Yeah I preferred to go safe and assume 0 is somehow possible.